src/input/keyboard.rs

```diff
@@ -11,38 +11,33 @@ use app::App;
 
 use glutin;
 
-#[derive(Clone, Debug)]
+#[derive(Debug, Copy, Clone)]
 pub struct ReceivedCharacter(pub char);
-#[derive(Clone, Debug)]
+#[derive(Debug, Copy, Clone)]
 pub struct KeyboardInput(pub glutin::ElementState, pub glutin::ScanCode, pub Option<glutin::VirtualKeyCode>);
-#[derive(Debug)]
+#[derive(Debug, Copy, Clone)]
 pub struct WidgetKeyboardInput(pub glutin::ElementState, pub glutin::ScanCode, pub Option<glutin::VirtualKeyCode>);
-#[derive(Debug)]
+#[derive(Debug, Copy, Clone)]
 pub struct WidgetReceivedCharacter(pub char);
 
-/**
-Note on focus:
-The tab key iterates through the widgets that have registered as focusable.
-Currently the order of this iteration is just based on the order the widgets
-are registered as focusable.
-Later on maybe it should be based on the relative positioning of widgets (could get
-ugly updating the treemap as widgets change position), or some user defined ordering.
-*/
+
+/// Note on focus:
+/// The tab key iterates through the widgets that have registered as focusable.
+/// Currently the order of this iteration is just based on the order the widgets
+/// are registered as focusable.
+/// Later on maybe it should be based on the relative positioning of widgets (could get
+/// ugly updating the treemap as widgets change position), or some user defined ordering.
+#[derive(Default)]
 pub struct FocusHandler {
     focusable_map: HashMap<WidgetRef, usize>,
-     // can replace TreeMap with std BTreeMap once the range API or similar is stable
+    /// Note: can replace TreeMap with std BTreeMap once the range API or similar is stable
```

Comment:
not all comments need to be doc comments, this one for instance is far more relevant while reading source than reading docs